circuits/types/field/src/from_bits.rs

```diff
@@ -31,7 +32,45 @@ impl<E: Environment> FromBits for Field<E> {
             // Check if all excess bits are zero.
             let should_be_zero = bits_le[size_in_bits..].iter().fold(Boolean::constant(false), |acc, bit| acc | bit);
             // Ensure `should_be_zero` is zero.
-            E::assert_eq(E::zero(), should_be_zero);
+            // TODO: `E::assert_eq` does not enforce constant constraints.
```

Comment:
This is intentional behavior, remove the comment

Reply:
What was the rationale for making it intentional?
Personally, I've run into this problem a few times while writing circuits.
I figure some sort of warning would help prevent errors.